influxql/ast.go

```diff
@@ -642,6 +642,36 @@ type SelectStatement struct {
 	FillValue interface{}
 }
 
+// HasDerivative returns true if one of the field in the statement is a
+// derivative aggregate
+func (s *SelectStatement) HasDerivative() bool {
```

Comment:
@otoolep this isn't correct as is.  It needs to check for `derivative` name on what `s.FunctionCalls()` returns... not fields.